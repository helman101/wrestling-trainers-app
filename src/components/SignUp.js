import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast, ToastContainer } from 'react-toastify';
import { userCreate } from '../API/api';
import styles from '../assets/styles/style.module.css';

const SignUp = (props) => {
  const [userName, setUserName] = useState('');
  const [userPassword, setUserPassword] = useState('');
  const [userEmail, setEmail] = useState('');
  const [userPConfirm, setUserPConfirm] = useState('');
  const handleChange = (e) => {
    const { name, value } = e.target;

    switch (name) {
      case 'name': {
        setUserName(value);
        break;
      }
      case 'email': {
        setEmail(value);
        break;
      }
      case 'password': {
        setUserPassword(value);
        break;
      }
      default: {
        setUserPConfirm(value);
      }
    }
  };

  const handleSubmit = () => {
    const params = {
      name: userName,
      email: userEmail,
      password: userPassword,
      passwordConfirm: userPConfirm,
    };
    props.dispatch(userCreate(params, props.history.push, toast));
  };

  return (
    <div className={`${styles.login}`}>
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />
      <div className={`${styles.sessionTitle}`}>SignUp</div>
      <div className={`${styles.mt2} ${styles.form} ${styles.dFlex} ${styles.flexColumn} ${styles.justifyContentCenter}`}>
        <input type="text" id="name" name="name" placeholder="Name" onChange={handleChange} className={`${styles.mb1}`} />
        <input type="email" id="email" name="email" placeholder="Email" onChange={handleChange} className={`${styles.mb1}`} />
        <input type="password" id="password" name="password" placeholder="Password" onChange={handleChange} className={`${styles.mb1}`} />
        <input type="password" id="passwordConfirm" name="passwordConfirm" placeholder="Confirm Password" onChange={handleChange} />
        <button type="button" onClick={handleSubmit} className={`${styles.mt2} ${styles.formBtn}`}>Submit</button>
      </div>
    </div>
  );
};

SignUp.propTypes = {
  dispatch: PropTypes.func.isRequired,
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
};

export default connect()(SignUp);
