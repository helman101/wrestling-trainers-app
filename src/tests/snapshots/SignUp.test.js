import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import defaultState from '../../store/store';
import SignUp from '../../components/SignUp';

const mockStore = configureStore([]);

describe('SignUp', () => {
  let store;
  let component;
  let history;
 
  beforeEach(() => {
    history = createMemoryHistory();
    store = mockStore(defaultState);
    component = renderer.create(
      <Provider store={store}>
        <SignUp history={history} />
      </Provider>
    );
  });
 
  it('should render propertly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});