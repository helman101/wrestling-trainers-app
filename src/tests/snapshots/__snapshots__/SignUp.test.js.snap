// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render propertly 1`] = `
<div
  className="login"
>
  <div
    className="Toastify"
  />
  <div
    className="sessionTitle"
  >
    SignUp
  </div>
  <div
    className="mt2 form dFlex flexColumn justifyContentCenter"
  >
    <input
      className="mb1"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    <input
      className="mb1"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <input
      className="mb1"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <input
      id="passwordConfirm"
      name="passwordConfirm"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
    />
    <button
      className="mt2 formBtn"
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;
